network.go

```diff
@@ -0,0 +1,365 @@
+	link, err := netlink.LinkByName(tapInterface.Name)
+	if err != nil {
+		return fmt.Errorf("LinkByName %s %v", tapInterface.Name, err)
+	}
+
+	tapl, ok := link.(*netlink.Tuntap)
+	if !ok {
+		return fmt.Errorf("incorrect interface type %s %v", tapInterface.Name, link)
+	}
+
+	if tapl == nil || tapl.Index == 0 {
+		return fmt.Errorf("delete tap interface not initialized")
+	}
+
+	err = netlink.LinkDel(tapl)
+	if err != nil {
+		return fmt.Errorf("destroy bridge %v", err)
+	}
+
+	*tapInterface = NetworkInterface{}
+
+	return nil
+}
+
+func createBridge() (*netlink.Bridge, string, error) {
+	linkName, err := genRandomInterface("bridge")
+	if err != nil {
+		return nil, "", err
+	}
+
+	bridge := &netlink.Bridge{LinkAttrs: netlink.LinkAttrs{Name: linkName}}
+
+	err = netlink.LinkAdd(bridge)
+	if err != nil {
+		return nil, "", fmt.Errorf("link add %s %v", linkName, err)
+	}
+
+	link, err := netlink.LinkByName(linkName)
+	if err != nil {
+		return nil, "", fmt.Errorf("LinkByName %s %v", linkName, err)
+	}
+
+	brl, ok := link.(*netlink.Bridge)
+	if !ok {
+		return nil, "", fmt.Errorf("incorrect interface type %v %v", linkName, link)
+	}
+
+	return brl, linkName, nil
+}
+
+func deleteBridge(brl *netlink.Bridge) error {
+	if brl == nil || brl.Index == 0 {
+		return fmt.Errorf("destroy bridge unnitialized")
+	}
+
+	err := netlink.LinkDel(brl)
+	if err != nil {
+		return fmt.Errorf("destroy bridge %v", err)
+	}
+
+	return nil
+}
+
+func enableBridge(brl *netlink.Bridge) error {
+	if brl == nil || brl.Index == 0 {
+		return fmt.Errorf("enable bridge unnitialized")
+	}
+
+	err := netlink.LinkSetUp(brl)
+	if err != nil {
+		return fmt.Errorf("enable link set up %v", err)
+	}
+
+	return nil
+}
+
+func disableBridge(brl *netlink.Bridge) error {
+	if brl == nil || brl.Index == 0 {
+		return fmt.Errorf("disable bridge unnitialized")
+	}
+
+	err := netlink.LinkSetDown(brl)
+	if err != nil {
+		return fmt.Errorf("disable link set down %v", err)
+	}
+
+	return nil
+}
+
+func getBridgeByName(linkName string) (*netlink.Bridge, error) {
+	if linkName == "" {
+		return nil, fmt.Errorf("unnamed bridge")
+	}
+
+	link, err := netlink.LinkByName(linkName)
+	if err != nil {
+		return nil, fmt.Errorf("link by name %v", err)
+	}
+
+	brl, ok := link.(*netlink.Bridge)
+	if !ok {
+		return nil, fmt.Errorf("incorrect interface type %s %v", linkName, link.Type())
+	}
+
+	return brl, nil
+}
+
+func addTAPInterfaceToBridge(tapInterface NetworkInterface, brl *netlink.Bridge) error {
+	link, err := netlink.LinkByName(tapInterface.Name)
+	if err != nil {
+		return fmt.Errorf("link by name %v", err)
+	}
+
+	tapl, ok := link.(*netlink.Tuntap)
+	if !ok {
+		return fmt.Errorf("incorrect interface type %s %v", tapInterface.Name, link.Type())
+	}
+
+	err = netlink.LinkSetMaster(tapl, brl)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func addVirtInterfaceToBridge(virtInterface NetworkInterface, brl *netlink.Bridge) error {
```

Comment:
There should not be any difference between adding a TAP and a regular interface to a bridge.